src/macros.jl

```diff
@@ -436,6 +436,10 @@ end
 
 # three-argument build_constraint is used for two-sided constraints.
 build_constraint(_error::Function, func::AbstractJuMPScalar, lb::Real, ub::Real) = build_constraint(_error, func, MOI.Interval(lb, ub))
+# To create `MOI.SingleVariable`-in-`MOI.Interval`, use
+# `@constraint(model, var in MOI.Interval(lb, ub)`.
+# See https://github.com/JuliaOpt/JuMP.jl/issues/1586
```

Comment:
This comment could be expanded to say that it intercepts `@constraint(model, lb <= var <= ub)` and promotes `var` to an `AffExpr` to form a `ScalarAffineFunction-in-Interval` instead of `SingleVariable-in-Interval`.

The multiplication by `1.0` is a bit subtle at first glance.